include/uv.h

```diff
@@ -1347,6 +1347,13 @@ UV_EXTERN uv_pid_t uv_os_getppid(void);
 UV_EXTERN int uv_os_getpriority(uv_pid_t pid, int* priority);
 UV_EXTERN int uv_os_setpriority(uv_pid_t pid, int priority);
 
+typedef struct {
+  long long quota_per_period;
+  long long period_length;
+  double proportions;
+} uv__cpu_constraint;
```

Comment:
How do I include properly while being able to test it in test-platform-output.c ?

Reply:
Take a look at e.g. test/test-idna.c, we do that by #including source files directly. You may need to split it off into a separate source file in order for that to work; in that case remember to update both CMakeLists.txt and Makefile.am.

Reply:
I have different leads to implement the test
- Either refactor `uv__get_constrained_cpu` into a separate `src/linux-cgroup.c` for instance, but my issue is my code depends on `uv__slurp` so I would need to extract it as well. Would it make sense ?
- I tried to include `internal.h` or `uv-common.h` in the test but the linker does not find `uv__get_constrained_cpu`, like src/unix/linux.c is not link for building test, am I wrong ?
- Third option is to drop the test

I am no C expert, so any help is welcome

Reply:
What about making `uv_get_constrained_cpu` an external definition just like `uv_get_constrained_memory`. It would simplify the tests + align with the memory implementation of cgroups

I would suggest the following documentation

```rst
.. c:function:: uv_cpu_constraint uv_get_constrained_cpu(void)

    Retrieves the CPU constraints imposed on the process by the operating system
    or container management system (e.g., cgroups on Linux). This function
    returns a `uv_cpu_constraint` structure that contains the details of the
    CPU constraints.

    If there are no CPU constraints, or if the constraints are unknown, the
    fields of the returned structure are set to zero. If there is a constraining
    mechanism but no specific constraints are set, the relevant fields are set
    to their maximum possible values.

    .. versionadded:: XXXX

.. c:type:: uv_cpu_constraint

    A structure representing the CPU constraints of a process.

    :c:member:`long long quota_per_period`
        The maximum amount of CPU time (in microseconds) that the process can
        consume in a given period. If the value is `0`, there is no quota
        enforced. If the value is `LLONG_MAX`, the quota is effectively
        unlimited.

    :c:member:`long long period_length`
        The length of the period (in microseconds) for which the quota is
        measured. A shorter period with the same quota allows for more
        fine-grained control of the process's CPU usage.

    :c:member:`double proportions`
        The weight or proportion of CPU resources available to the process
        relative to other processes. This is a normalized value, typically
        between `0.0` and `1.0`, where `1.0` represents full access to a single
        CPU core, and `0.5` would represent half of a CPU core. This value is
        used in conjunction with `quota_per_period` and `period_length` to
        determine the actual CPU time available to the process.
```

And update `uv_available_parallelism` documentation accordingly

What do you think ?

Reply:
I don't want to extend the public API on a whim because chances are good that without careful discussion and deliberation, it gets overfitted to the thing at hand (i.e., cgroups.)

You can open an issue to discuss but consider it out of scope for this pull request for now.